### ISSUE TITLE
Log the current channel when logging HTTP/2 frames

### ISSUE THREAD

**Issue by trustin** [netty/netty#4429]
When running concurrent HTTP/2 connections, it's hard/impossible to tell which frame is from which connection. `Http2FrameLogger` should log the current connection as well.

Also, please consider aesthetics. The first line with the hyphens and the last line with the hyphens end at the different column for example. Could we get something nicer such as we have with `LoggingHandler`?

**Comment by Scottmitch** [comment]
no objection here. were you going to submit a pr @trustin? The unknown frame type may require a bit more than just a set lookup.

**Comment by ifesdjeen** [comment]
Do you think something like Channel Id might be helpful there?..

**Comment by trustin** [comment]
Just passing the channel itself to the logger should work:

    logger.debug("{}", channel);

**Comment by ifesdjeen** [comment]
I can work on the patch of nobody has it in the  pipeline.

**Comment by normanmaurer** [comment]
Yes please. Thanks buddy!

**Comment by ifesdjeen** [comment]
It's going to look something like that: 

```
DATA: channel=[id: 0x7f64c45e, /127.0.0.1:50317 => /127.0.0.1:8080], streamId=3, padding=0, endStream=true, length=24, bytes=48656c6c6f20576f726c64202d2076696120485454502f32
```

Do you think it's helpful?
Although I assume it's undesireable to change an API of `Http2FrameLogger`, right? Should we just duplicate the methods or is it assumed to be used only in `Inbound`/`Outbound` frame loggers? WDYT?

**Comment by trustin** [comment]
@ifesdjeen Yes, it's gonna be helpful. I think the log message could look similar to that of `LoggingHandler` for consistency though. Also, we never released HTTP/2 in 4.0, so it's OK to make a breaking change.

**Comment by Scottmitch** [comment]
@ifesdjeen - I think you submitted a PR for this ... can this be closed?

**Comment by ifesdjeen** [comment]
@Scottmitch the issue with channel is solved. I'll work on the patch for the second part @trustin suggested during the week, but it's more of a separate issue I think.

**Comment by Scottmitch** [comment]
@ifesdjeen - I see this is kinda a 2 part issue. Lets leave it open until `Http2FrameLogger` is consistent with `Http2FrameLogger`. Thanks for taking this on!

**Comment by Scottmitch** [comment]
@trustin @ifesdjeen - can you comment on the progress for the second part of this issue?

**Comment by ifesdjeen** [comment]
@Scottmitch sorry I've been quite busy lately, couldn't finish it. Is it still releavant for me to try and do it or you've already taken over?

**Comment by Scottmitch** [comment]
@ifesdjeen - If you have bandwidth please feel free to tackle this. If not let me know (I haven't taken any action on this).
